include/TFragment.h

```diff
@@ -25,96 +25,75 @@
-   UShort_t NumberOfPileups;      //-> Number of piled up hits 1-3
-   UShort_t DataType;             //-> 
-   UShort_t DetectorType;         //-> Detector Type (PACES,HPGe, etc)
-   UInt_t ChannelId;              //-> Threshold crossing counter for a channel
-   //UInt_t AcceptedChannelId;      //-> Accepted threshold crossing counter for a channel
-
-   std::vector<UShort_t>  KValue; //-> KValue for each pileup hit
-
-   /// *****************************  ////
-
-   std::vector<Short_t>  wavebuffer;//-> waveform words
-
-   TPPG* fPPG; //!<!
-
-	int NumberOfHits;            //!<! transient member to count the number of pile-up hits in the original fragment
-	int HitIndex;                //!<! transient member indicating which pile-up hit this is in the original fragment
- 
-   UInt_t GetAddress()   const { return ChannelAddress; }
-   double GetTime()      const; //!<!
-   long   GetTimeStamp() const; //!<!
-   double GetTZero() const; //!<!
-   const char *GetName() const; //!<!
-   double GetEnergy(size_t iter=0)const; //!<!
-   Float_t GetCharge(size_t iter=0)const; //!<!
-   long GetTimeStamp_ns() const; //!<!
-   ULong64_t GetTimeInCycle(); //!<!
-   ULong64_t GetCycleNumber(); //!<!
-
-	time_t GetMidasTimeStamp() const { return MidasTimeStamp; }  //!<!
-	Short_t GetChannelNumber() const { return ChannelNumber; }  //!<!
-   Int_t GetCfd(size_t iter=0) const { return Cfd.at(iter); }  //!<!
-   Int_t GetZc(size_t iter=0) const  { if(iter < Zc.size()) return Zc.at(iter); return 0; } //!<! 
-   Int_t GetLed(size_t iter=0) const { return Led.at(iter); }  //!<!
-   Int_t GetCcShort(size_t iter=0) const { if(iter < ccShort.size()) return ccShort.at(iter); return 0; }  //!<!
-   Int_t GetCcLong(size_t iter=0) const { if(iter < ccLong.size()) return ccLong.at(iter); return 0; }  //!<!
-	UShort_t GetDeadTime() const { return DeadTime; }  //!<!
-	UInt_t GetChannelId() const { return ChannelId; }  //!<!
-
-   Int_t Get4GCfd(size_t i=0) const; //!<!
-
-   bool IsDetector(const char *prefix, Option_t *opt = "CA") const; //!<!
-   int  GetColor(Option_t *opt = "") const; //!<!
-   bool HasWave() const { return (wavebuffer.size()>0) ?  true : false; } //!<!
-
-   virtual void	Clear(Option_t *opt = ""); //!<!
-   virtual void Print(Option_t *opt = "") const; //!<!
-  
-   virtual Int_t  GetDetector() const; //!<!
-   virtual Int_t  GetSegment() const;	 //!<!
-
-
-
-
-   bool operator<(const TFragment& rhs) const { return (GetTimeStamp() < rhs.GetTimeStamp()); }
-   bool operator>(const TFragment& rhs) const { return (GetTimeStamp() > rhs.GetTimeStamp()); }
-
-/// \cond CLASSIMP
-   ClassDef(TFragment,5);  // Event Fragments
-/// \endcond
+class TFragment : public TGRSIDetectorHit	{
+  public:
+    TFragment(); 
+    TFragment(const TFragment&); 
+    virtual ~TFragment(); 
+
+    time_t   MidasTimeStamp;       //->  Timestamp of the MIDAS event  
+    Int_t    MidasId;              //->  MIDAS ID
```

Comment:
So this adds a whole bunch of stuff to the hit that wasn't there previously. Are we happy with that?